packages/web-client/tests/acceptance/pay-test.ts

```diff
@@ -192,6 +229,49 @@ module('Acceptance | pay', function (hooks) {
     assert.dom(PAYMENT_URL).containsText(expectedUrl);
   });
 
+  test('it renders correctly if currency is non-USD and non-SPEND', async function (assert) {
+    const jpyAmount = 300.335;
+    const jpySymbol = 'JPY';
+    const roundedJpyAmount = roundAmountToNativeCurrencyDecimals(
+      jpyAmount,
+      jpySymbol
+    );
+
+    await visit(
+      `/pay/${network}/${merchantSafe.address}?amount=${jpyAmount}&currency=${jpySymbol}`
+    );
+    await waitFor(MERCHANT);
+
+    assert.dom(MERCHANT).hasAttribute('data-test-merchant', merchantName);
+    assert
+      .dom(MERCHANT_LOGO)
+      .hasAttribute(
+        'data-test-merchant-logo-background',
+        merchantInfoBackground
+      );
+    assert
+      .dom(MERCHANT_LOGO)
+      .hasAttribute(
+        'data-test-merchant-logo-text-color',
+        merchantInfoTextColor
+      );
+
+    assert
+      .dom(AMOUNT)
+      .containsText(convertAmountToNativeDisplay(roundedJpyAmount, jpySymbol));
+    assert.dom(SECONDARY_AMOUNT).doesNotExist();
+
+    let expectedUrl = generateMerchantPaymentUrl({
+      domain: universalLinkDomain,
+      network,
+      merchantSafeID: merchantSafe.address,
+      currency: jpySymbol,
+      amount: Number(roundedJpyAmount),
+    });
+    assert.dom(QR_CODE).hasAttribute('data-test-styled-qr-code', expectedUrl);
+    assert.dom(PAYMENT_URL).containsText(expectedUrl);
+  });
```

Comment:
I don’t know where this belongs exactly (probably not this PR!) but I’ve never seen this representation of the local currency:

![image](https://user-images.githubusercontent.com/43280/134017744-5bf49229-5d45-46a2-b3ac-f646b0fa4fe2.png)

It would usually be `C$100` or `$100 CAD` but not both (nor is `CA` really a thing when it comes to currency, that I’ve seen)

Reply:
This is leftover from rainbow. Let's change it if we can settle on a better way to represent it (and if it doesn't break any assumptions of mobile design), issue in CS-1897.